packages/lit-html/src/test/lit-html_test.ts

```diff
@@ -1469,4 +1469,81 @@ suite('lit-html', () => {
       assert.isOk(event);
     });
   });
+
+  class DisconnectingDirective extends Directive {
+    log!: Array<string>;
+
+    render(log: Array<string>) {
+      this.log = log;
+      return 'hello';
+    }
+
+    disconnectedCallback() {
+      this.log.push('disconnected');
+    }
+  }
+  const disconnectingDirective = directive(DisconnectingDirective);
+
+  // TODO (justinfagnani): dynamic directives are not handled correctly in
+  // general. We may not even want to support them. TBD.
```

Comment:
I didn't follow, what isn't handled correctly here? We [dirty-check the constructor value](https://github.com/Polymer/lit-html/pull/1216/files#diff-073dbf13ebed6ac6ef2cce7a849a36a3R599) to know whether to clear the node part (which should disconnect the previous directive); committing any other value type (like `nothing` here) should also cause a clear. What am I missing?

Reply:
I hit this in `templateContent` and filed an issue: https://github.com/Polymer/lit-html/issues/1286